### ISSUE TITLE
[A11y Docs] [AnchoredOverlay] component

### ISSUE THREAD

**Issue by patrickhlauke** [patrickhlauke/kamino#33]
**[<img src="https://avatars.githubusercontent.com/u/895831?s=17&v=4" width="17" height="17"> @patrickhlauke](https://github.com/patrickhlauke)** cloned issue [TetraLogical/GitHub-Primer-06-2024#107](https://github.com/TetraLogical/GitHub-Primer-06-2024/issues/107) on 2024-07-02: 

> ## Documentation page
> 
> 
> 
> * https://primer.style/components/anchored-overlay
> 
> 
> 
> ## Description
> 
> 
> 
> As part of our current review of all Primer components, we suggest making the following changes/additions to the documentation for this component (as a continuation of the activity started in https://github.com/github/primer/issues/2966).
> 
> 
> 
> Note that the below are initial thoughts and suggestions, based on the structure used for the previous documentation effort. It does not represent the final wording to be used/merged into the documentation.
> 
> 
> 
> We further recommend first addressing/fixing any bugs and issues identified for the component before updating the documentation.
> 
> 
> 
> ---
> 
> 
> 
> ## Additions/changes to the component's existing Usage description (optional)
> 
> 
> 
> > If this is not already sufficiently covered in the existing component documentation, make sure to explain at a high level what the component should be used for.
> 
> 
> 
> The main overview documentation should mirror/copy the documentation for the "regular" overlay, and give examples of when authors may wish to consider an overlay - and when other components, such as dialogs, may be more appropriate.
> 
> 
> 
> The documentation briefly mentions, under the current accessibility section, that "The anchored overlay menu will automatically receive a focus trap and focus zone. Use up/down keys to navigate between buttons". As the documentation does not actually explain how an anchored overlay can be used for menus, this information is not very useful.
> 
> 
> 
> The Storyboard examples for the component are minimal and broken (the placeholder image used in the examples does not load). They are also not representative of the various use cases for the overlay (including the use as a menu, as mentioned above). We recommend creating new, representative examples that closely match real-world uses for the component.
> 
> 
> 
> ## Accessibility and usability expectations
> 
> 
> 
> > What is the expected "ideal" form for the component. This includes aspects of what would normally be more formal User Acceptance Testing requirements, but written in a less dry/bullet list way, and more as an explainer in prose.
> 
> > List the essential accessibility aspects for the component (e.g. "the control must have an accessible name", "the selected item must correctly expose its state", "focus must be moved into the component when it is opened").
> 
> > Avoid simply listing WCAG Success Criteria - while it's ok to mention specific SCs, this section should not be a simple checklist of SCs.
> 
> > List the expected interaction methods for the component - how it should react to mouse/touch interactions, what the expected keyboard experience is, what the outcome of an interaction should be (e.g. opening a new window, closing the widget, refreshing the current page).
> 
> > If useful/necessary, create subsections/subheadings within this more freeform section.
> 
> > Add a boilerplate reference/link to the more general/generic accessibility requirements that apply to all components in general https://primer.style/guides/accessibility
> 
> 
> 
> ## Built-in accessibility features
> 
> 
> 
> > Aimed at developers that "pick up" the component, this section outlines - of all the "ideal" form expectations listed above - what the component gives them out-of-the-box, for free (e.g. if the component already comes with ready-made focus handling, `aria-live` notifications, relevant keyboard shortcuts).
> 
> > Point out how authors should be using/instantiating components correctly to get those built-in benefits (e.g. which attributes/properties *must* be used in order to guarantee that the component is baseline accessible).
> 
> > If useful/necessary, create subsections/subheadings within this more freeform section.
> 
> > Provide live code samples to demonstrate good/correct use of the component (ideally embedded directly in the documentation itself), calling out/demonstrating any specific features of note mentioned here.
> 
> 
> 
> ## Implementation requirements
> 
> 
> 
> > Aimed at developers, this section explains any dependencies or aspects that authors *must* take care of themselves, explicitly, in order to guarantee that the result is accessible. (e.g. clearly explaining if a component only receives an accessible name if combined with another "labelling" component, and then only if the component is then given an `aria-labelledby` pointing to the `id` of the latter; the need to include a separate notification component in order to dynamically announce aspects of the component to screen readers).
> 
> > If useful/necessary, create subsections/subheadings within this more freeform section.
> 
> > Provide live code samples to demonstrate good/correct use of the component with the addition of extra work/dependencies that authors need to handle themselves (ideally embedded directly in the documentation itself).
> 
> 
> 
> ## How to test the component
> 
> 
> 
> ### Integration tests
> 
> 
> 
> > Tests specifically aimed at authors who integrate the component in their own work
> 
> 
> 
> ### Component tests
> 
> 
> 
> > Tests specifically aimed at component authors/maintainers - general acceptance criteria/tests to verify the component does what it's supposed to
> 
> 
> 
>
